### ISSUE TITLE
Print a warning/throw an error when params in optimizer are not the same as that of module's in `make_private`

### ISSUE THREAD

**Issue by karthikprasad** [pytorch/opacus#432]
...code is failing because `model = ModuleValidator.fix(model)` is called after passing `model.parameters()` to optimizer. 
`ModuleValidator.fix(model)` changes some layers of the model (specifically replaces `BatchNorm` with `GroupNorm`) and the parameters of these new layers are not tracked by optimizer.

Calling `fix()` before defining optimizer should work.

That said, your issue does create two action items for us:
1. At the very least, print a warning/throw an error when params in optimizer are not the same as that of modules.
2. Print a more helpful message in case of failures. (not sure how)

_Originally posted by @karthikprasad in https://github.com/pytorch/opacus/issues/431#issuecomment-1136349814_
